### ISSUE TITLE
Hessian non-diagonal elements?

### ISSUE THREAD

**Issue by Guruprasad93** [Ageliss/For_shared_codes#1]
Hi,

If I understand right, your script only gives the diagonal terms of the hessian matrix and not the off-diagonal terms. Is there an efficient strategy to compute the off-diagonal terms of the hessian for any neural net?

**Comment by Ageliss** [comment]
Yes, it provides only with the diagonal terms. I don't think Pytorch gives a more effecient way to calculate this hessian matrix. Instead, TF gives an efficient way. Try below referenced from https://www.tensorflow.org/tutorials/customization/autodiff: 
'''
x = tf.Variable(1.0)  # Create a Tensorflow variable initialized to 1.0

with tf.GradientTape() as t:
  with tf.GradientTape() as t2:
    y = x * x * x
  # Compute the gradient inside the 't' context manager
  # which means the gradient computation is differentiable as well.
  dy_dx = t2.gradient(y, x)
d2y_dx2 = t.gradient(dy_dx, x)

assert dy_dx.numpy() == 3.0
assert d2y_dx2.numpy() == 6.0
'''
